### PR TITLE
78 update the tests for app companies

### LINKED ISSUE #78: Update the tests for app 'companies'
- Rewrite the tests due to changes in the authentication system.
- Сorrect the URLs for the tests.
- Correct the company data validation in the serializer.

### PR DESCRIPTION
**Pull request by YanZhylavy**
- The tests for app 'companies' have been fixed according to the new authentication system and updated URLs
- Validation methods in the serializer have been corrected

### CODE REVIEW

File: companies/tests.py [hunk 1]
```diff
@@ -1,68 +1,80 @@
...
+from forum.settings import FRONTEND_URL
 
 
 class CompanyTestAuthenticatedUser(APITestCase):
 
     def setUp(self):
-        self.company_url = reverse('companies-list-create')
-        user = CustomUser.objects.create_user('test@mail.com','Test_password123456')
+        self.company_url = f'{FRONTEND_URL}/companies/'
+        self.get_company_url =  f'{self.company_url}get_company/'
```
Comment (akostromin14) [inline companies/tests.py:12]: U can use reverse method for simplify url work, what do you think?

File: companies/tests.py [hunk 2]
```diff
@@ -1,68 +1,80 @@
...
     def test_valid_edrpou(self):
-        valid_post_body = {'brand': 'test_brand', 'is_registered': True, 'edrpou': 98756775}
+        valid_post_body = {'brand': 'test_brand', 'is_startup': True, 'edrpou': 98756775}
         response = self.client.post(self.company_url, valid_post_body, format='json')
         self.assertEqual(response.status_code, 201)
 
     def test_valid_phone(self):
-        valid_post_body = {'brand': 'test_brand', 'is_registered': True, 'contact_phone': '+380937777777'}
+        valid_post_body = {'brand': 'test_brand', 'is_startup': True, 'contact_phone': '+380937777777'}
         response = self.client.post(self.company_url, valid_post_body, format='json')
```
Comment (akostromin14) [inline companies/tests.py:59]: Next time try to use faker but now we haven't got time for that.

File: validation/serializers.py [hunk 3]
```diff
@@ -13,16 +13,18 @@ def validate_edrpou(value):
         return value
     
     @staticmethod
-    def validation_email(email_to_validate: str):
+    def validate_contact_email(email_to_validate: str):
         pattern = r'([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+'
         if not re.match(pattern, email_to_validate):
             raise ValidationError("Email is invalid")
+        return email_to_validate
```
Comment (akostromin14) [inline validation/serializers.py:20]: Do we use this method in the project?
We have to rename it everywhere, check please.

**Review by akostromin14 — APPROVED** [review]
Good work!

File: validation/serializers.py (hunk 3 above)
Comment (YanZhylavy) [reply validation/serializers.py:20]: This method is automatically invoked by the serializer, DRF does this using the pattern validate_"fieldname".

File: companies/tests.py (hunk 1 above)
Comment (YanZhylavy) [reply companies/tests.py:12]: unfortunatly, here DRF's viewset and router are used, so reverse() doesn't realy works

**Review by mehalyna — APPROVED** [review]

**Pull request merged**